src/modsim2/similarity/metrics/pad.py

```diff
@@ -0,0 +1,310 @@
+    kernels allowed in sklearn's SVC, and the data can be transformed
+    with any of the embeddings available in this project. Multiple
+    classifiers may be created to allow for some hyperparameter tuning.
+
+    """
+
+    def __init__(self, seed: int):
+        super().__init__(seed)
+
+        self.test_proportion = 0.2  # The % of the dataset to be held out for testing
+        self.train_proportion_equal = (
+            False  # Boolean to determine if train datasets should be equal size
+        )
+        self.embedding_name = None  # Name of the embedding function
+        self.train_data = None
+        self.train_labels = None
+        self.test_data = None
+        self.test_labels = None
+        self.embed_train_data = None
+        self.embed_test_data = None
+        self.models = []  # List of trained classifiers
+        self.errors = []  # List of errors for classifiers on test data
+
+    def __train_test_split(
+        self,
+        data_A: np.ndarray,
+        data_B: np.ndarray,
+    ):
+        """
+        This method splits the two datasets data_A and data_B into train and test
+        samples. To calculate the PAD, there should be the same number of samples
+        from A & B in the test dataset. However, there may not be the same number
+        of samples in A & B. The following lines of code determine the size of the
+        test dataset for both A & B - the minimum proportion size of the two datasets
+
+        Args:
+            data_A: all records in dataset A (excludes target value)
+            data_B: all records in dataset B (excludes target value)
+
+        Returns:
+            train_A: records in dataset A to be used for training
+            test_A: records in dataset A to be used for testing (final evaluation)
+            train_B: records in dataset B to be used for training
+            test_B: records in dataset B to be used for testing (final evaluation)
+        """
+        # Determine the number of samples of test data to be drawn from A&B
+        test_size_A = np.round(data_A.shape[0] * self.test_proportion)
+        test_size_B = np.round(data_B.shape[0] * self.test_proportion)
+        test_size = int(min(test_size_A, test_size_B))
+        # Split A&B into train and test
+        train_A, test_A = train_test_split(data_A, test_size=test_size, shuffle=True)
+        train_B, test_B = train_test_split(data_B, test_size=test_size, shuffle=True)
+
+        if self.train_proportion_equal:
+            # If the train_A and train_B have different numbers of records then drop
+            # records from the larger of the two datasets so that A & B are of
+            # equal size
+            if train_A.shape[0] > train_B.shape[0]:
+                train_A = self._reduce_size_array(train_A, train_B.shape[0])
+            elif train_B.shape[0] > train_A.shape[0]:
+                train_B = self._reduce_size_array(train_B, train_A.shape[0])
```

Comment:
I like the option to use an equal amount of data from A and B in the train set, but I think the alternative (where data are imbalanced) should be to preserve the original balance of the data.

So, if A is 2/3 and B is 1/3, then the train set should be either 2/3 A and 1/3 B, or 50/50. The current setup makes the imbalance marginally larger by default, which I don't think is an option we should use. I don't necessarily think it will make a huge difference, just that either preserving balance or going 50/50 should be the two valid options!

Reply:
Thanks - I misunderstood the original comments. I have created two options 'equal' and 'ratio' where there is either a 50:50 split between A&B in the training data or the original ratio is maintained.